### PR TITLE
Fix for adapter access revert

### PR DESCRIPTION
**Pull request by johnyrahul**
## What

The adapter access revert was failing.

## Why

- when we try to revert adapter access of certain users, we do additional logic to remove those adapters from the user's default triad if the user has set the shared adapter as his default. However, there can be scenarios where the user has not configured the default adapter.

## How

- Added try catch and continue the iteration if there is no entry in the UserDefaultAdapter model.

## Notes on Testing

-

## Checklist

I have read and understood the [Contribution Guidelines]().

### CODE REVIEW

File: backend/adapter_processor/views.py [hunk 1]
```diff
@@ -247,20 +247,33 @@ def partial_update(
...
-                elif user_default_adapter.default_x2text_adapter == adapter:
-                    user_default_adapter.default_x2text_adapter = None
-
-                user_default_adapter.save()
+                try:
+                    user_default_adapter = UserDefaultAdapter.objects.get(
+                        user_id=user_id
+                    )
+
+                    if user_default_adapter.default_llm_adapter == adapter:
```
[the remark this answers is not in the document]
Comment (johnyrahul) [reply backend/adapter_processor/views.py:255]: @chandrasekharan-zipstack Is the below what you are expecting?

`if adapter.type == "LLM" and user_default_adapter.default_llm_adapter == adapter:`

[the remark this answers is not in the document]
Comment (chandrasekharan-zipstack) [reply backend/adapter_processor/views.py:255]: Yes @johnyrahul

File: backend/adapter_processor/views.py [hunk 2]
```diff
@@ -247,20 +247,33 @@ def partial_update(
...
+                    elif (
+                        user_default_adapter.default_vector_db_adapter
+                        == adapter
+                    ):
+                        user_default_adapter.default_vector_db_adapter = None
+                    elif user_default_adapter.default_x2text_adapter == adapter:
+                        user_default_adapter.default_x2text_adapter = None
+
+                    user_default_adapter.save()
+                except UserDefaultAdapter.DoesNotExist:
```
Comment (chandrasekharan-zipstack) [inline backend/adapter_processor/views.py:271]: NIT: This exception would be caught only when calling `UserDefaultAdapter.objects.get()`. We could move this check over there

**Review by gaya3-zipstack — APPROVED** [review]
Looks fine.

File: backend/adapter_processor/views.py (hunk 2 above)
Comment (johnyrahul) [reply backend/adapter_processor/views.py:271]: I wanted to use `user_default_adapter`  in the below line,  else I would be required to initialize it outside the try block with none. Also additionally will require to add a null check before accessing the elements inside the variable.

**Review by chandrasekharan-zipstack — APPROVED** [review]

**Pull request merged**